### ISSUE TITLE
[babel-plugin] in debug, styles values with dot are unsupported

### ISSUE THREAD

**Issue by hrasoa** [facebook/stylex#937]
### Describe the issue

<img width="654" alt="Image" src="https://github.com/user-attachments/assets/1ceaea50-d589-4ef8-a323-e30a7448c5f9" />

### Expected behavior

It should work.

### Steps to reproduce

create styles like:

```ts
export const styles = stylex.create({
  [1.5]: {
    paddingLeft: '1px',
  }
})
```
enable `debug` in the babel config.

**Comment by necolas** [comment]
The step to reproduce that you've included has to be wrong, because the screenshot shows a variable. You can avoid this issue by not using numbers as keys when using `defineVars`

**Comment by hrasoa** [comment]
oh yeah i completely messed up the repro, so I can't write tokens like [here](https://github.com/hrasoa/stylex-repro/blob/main/src/token.stylex.ts)?

```ts
export const tokens = stylex.defineVars({
  "1.5": "red",
});
```

**Comment by necolas** [comment]
No, `--1.5` is not a valid CSS custom property. The compiler should throw an error though, instead of generating the bad code

**Comment by necolas** [comment]
Why should we allow numbers or numeric strings?

**Comment by nmn** [comment]
@necolas It's easier to simply allow valid JS syntax than to enforce and document the limitations.

**Comment by necolas** [comment]
There are many things we need to enforce and document, so it's unfortunate we're adding more stuff like this instead. There's a general need to remove complexity and support for odd patterns, so let's make this the last of them. We can revisit it later
